prefect_azure/credentials.py

```diff
@@ -177,23 +184,25 @@ async def example_get_container_client_flow():
             ```
         """
         container_client = ContainerClient.from_connection_string(
-            self.connection_string, container
+            self.connection_string.get_secret_value(), container
         )
         return container_client
 
 
-@dataclass
-class CosmosDbAzureCredentials:
+class CosmosDbAzureCredentials(Block):
     """
-    Dataclass used to manage Cosmos DB authentication with Azure.
+    Block used to manage Cosmos DB authentication with Azure.
     Azure authentication is handled via the `azure` module through
     a connection string.
 
     Args:
         connection_string: includes the authorization information required
     """
 
-    connection_string: str
+    _block_type_name = "Cosmos DB Azure Credentials"
+    _logo_url = "https://github.com/PrefectHQ/orion/blob/main/docs/img/collections/azure.png?raw=true"  # noqa
```

Comment:
Starting with Azure makes sure that they'll all be next to each other in the blocks UI. Maybe update the class name too?
```suggestion
    _block_type_name = "Azure Cosmos DB Credentials"
    _logo_url = "https://images.ctfassets.net/gm98wzqotmnx/6AiQ6HRIft8TspZH7AfyZg/39fd82bdbb186db85560f688746c8cdd/azure.png?h=250"  # noqa
```